### ISSUE TITLE
"bun upgrade" used to work for me, but fails now

### ISSUE THREAD

**Issue by mvolkmann** [oven-sh/bun#11523]
### What version of Bun is running?

1.1.12+43f0913c3

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

Enter `bun upgrade`

### What is the expected behavior?

successful upgrade

### What do you see instead?

Bun upgrade failed with error: HTTPError

Please upgrade manually:
  curl -fsSL https://bun.sh/install | bash

**Comment by qinsong77** [comment]
same here

**Comment by mvolkmann** [comment]
@Jarred-Sumner This is still happening in Bun 1.1.13. Seems like a serious issue.

**Comment by balazs4** [comment]
same.

```sh
# bun upgrade
Bun upgrade failed with error: HTTPError

Please upgrade manually:
  curl -fsSL https://bun.sh/install | bash

# bun --version
1.1.17
```

**Comment by balazs4** [comment]
I think non-valid (e.g. expired) [GITHUB_TOKEN](https://github.com/oven-sh/bun/blob/2f0789af7c46d54e75c7d1639f2ddc19f9aeb7d6/src/cli/upgrade_command.zig#L215) might cause the issue.

```sh
# GITHUB_TOKEN=pat_invalid bun upgrade

Bun upgrade failed with error: HTTPError

Please upgrade manually:
  curl -fsSL https://bun.sh/install | bash
```

same with valid GITHUB_TOKEN

```sh
# bun upgrade
Congrats! You're already on the latest version of Bun (which is v1.1.17)
```

## workaround

you have to make sure GITHUB_TOKEN env either not defined or is valid where you want to run `bun upgrade`

```sh
GITHUB_TOKEN= bun upgrade
```

**Comment by mvolkmann** [comment]
YES! Removing that environment variable fixed it for me. I think this should be documented here: https://bun.sh/docs/installation#upgrading
and then this issue can be closed.
